packages/trpc/server/routers/viewer/workflows/update.handler.ts

```diff
@@ -262,7 +262,7 @@ export const updateHandler = async ({ ctx, input }: UpdateOptions) => {
         },
       });
 
-      const promiseSteps = steps.map(async (step) => {
```

Comment:
`steps` includes new steps that are not yet created in the DB. That's why it was throwing the foreign key error